packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1495,6 +1495,9 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   @override
   void hideToolbar() {
+    if (_selectionOverlay == null || !_selectionOverlay.toolbarIsVisible) {
```

Comment:
I am not sure if it is necessary, we might also want to check _selectionOverlay.handlesarevisible

Reply:
I just tried playing around with it and using `handlesAreVisible` and I think it has to stay as-is.  The use case is tapping to close the selection menu in an empty text field.  In that case no handles are shown, only the toolbar, but we still want to close the toolbar on tap.  However I also don't want to call `selectionOverlay.hide` when there is no selection overlay, like tapping on an empty field without the menu up.